lib/booking_system.rb

```diff
@@ -0,0 +1,82 @@
+require_relative 'date_range'
+require_relative 'reservation'
+require 'pry'
+
+module Hotel
+  class BookingSystem
+    attr_reader :reservations, :rooms
+
+    VALID_ROOM_NUMBERS = (1..20)
+    ROOM_COST = 200
+    DISCOUNT_ROOM_RATE = 150
+
+    def initialize
+      @reservations = []
+      @rooms = VALID_ROOM_NUMBERS
+    end
+
+
+    ####
+    # room: an Integer for the room number
+    # check_in: a Date object for the check in day
+    # check_out: a Date object for the check out day
+    def book_room(room_number, check_in, check_out)
+      unless VALID_ROOM_NUMBERS.include?(room_number)
+        raise ArgumentError
+      end
```

Comment:
Good use of vertical white space throughout this file - you do a good job of breaking up ideas, and it makes the whole thing much easier to read.